### PR TITLE
Fix distribcell hex lattice bug

### LINKED ISSUE #411: Distribcell tally reporting for nested hexagonal lattices is incorrect
Per [this](https://groups.google.com/forum/#!topic/openmc-users/PiTNpttWXOA) thread on the Google users group, there is a bug in the way that distribcell tallies are reported in "tallies.out" for nested hexagonal lattices. From my quick analysis, it appears that the distribcell algorithm counts and tallies the correct number cell instances throughout a nested hex geometry, but builds incorrect and inconsistent string "paths" for the "tallies.out" file for each instance. Hence, the bug is very likely in the ``find_offset(...)`` routine in [src/offset.F90](https://github.com/mit-crpg/openmc/blob/develop/src/output.F90#L2191).

I probably won't have time to look into this for two days. @smharper, do you have an hour or so to try to figure this one out? You may find the issue faster than I since hex lattice indexing is still a bit over my head.

### PR DESCRIPTION
**Pull request by smharper**
This closes #411.  We were just missing the check to see if the current lattice tile was the last.

I have verified that the correct tally results are output at the correct positions, and I've made a test so we'll see if this regresses in the future.

### CODE REVIEW

File: tests/test_filter_distribcell/case-4/results_true.dat [hunk 1]
```diff
@@ -0,0 +1,17 @@
+k-combined:
+0.000000E+00 0.000000E+00
+tally 1:
+2.166056E-02
```
Comment (wbinventor) [inline tests/test_filter_distribcell/case-4/results_true.dat]: Why are there results for 14 tally bins here? I would have though there would only be 7 for the 7 pin cells in the problem.

**Comment by wbinventor** [comment]
This look really good @smharper, thanks so much for digging into this. I'm breathing a sigh of relief right now that it wasn't a systematic problem with distribcells, just the hex lattice indexing for the final report!

File: tests/test_filter_distribcell/case-4/results_true.dat (hunk 1 above)
Comment (smharper) [inline tests/test_filter_distribcell/case-4/results_true.dat]: 14 bins = 7 sum(x) bins + 7 sum(x^2) bins

Comment (wbinventor) [inline tests/test_filter_distribcell/case-4/results_true.dat]: Oh wow, I've been thinking deterministic transport too much this summer. Of course!

**Pull request merged**